backend/attendance-service/src/main/java/org/egov/validator/StaffServiceValidator.java

```diff
@@ -1,11 +1,283 @@
 package org.egov.validator;
 
+import com.jayway.jsonpath.JsonPath;
+import digit.models.coremodels.RequestInfoWrapper;
 import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.ObjectUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.egov.common.contract.request.RequestInfo;
+import org.egov.service.AttendanceRegisterService;
+import org.egov.tracer.model.CustomException;
+import org.egov.util.MDMSUtils;
+import org.egov.web.models.AttendanceRegister;
+import org.egov.web.models.AttendanceRegisterSearchCriteria;
+import org.egov.web.models.StaffPermission;
+import org.egov.web.models.StaffPermissionRequest;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
+import org.springframework.util.CollectionUtils;
+
+import java.math.BigDecimal;
+import java.util.*;
+
+import static org.egov.util.AttendanceServiceConstants.MASTER_TENANTS;
+import static org.egov.util.AttendanceServiceConstants.MDMS_TENANT_MODULE_NAME;
+
 
 @Component
 @Slf4j
 public class StaffServiceValidator {
-    //TODO
+
+    @Autowired
+    private MDMSUtils mdmsUtils;
+
+    @Autowired
+    private AttendanceRegisterService attendanceRegisterService;
+
+
+
+    public void validateCreateStaffPermission(StaffPermissionRequest request,List<StaffPermission> staffPermissionListFromDB
+            ,List<AttendanceRegister> attendanceRegisterListFromDB) {
+        RequestInfo requestInfo=request.getRequestInfo();
+        List<StaffPermission> staffPermissionListFromRequest=request.getStaffPermissionList();
+        Map<String, String> errorMap = new HashMap<>();
+
+            String rootTenantId = staffPermissionListFromRequest.get(0).getTenantId();
+            //split the tenantId
+            rootTenantId = rootTenantId.split("\\.")[0];
+
+            Object mdmsData = mdmsUtils.mDMSCall(requestInfo, rootTenantId);
+
+            //check tenant Id
+            validateMDMSData(staffPermissionListFromRequest.get(0), mdmsData, errorMap);
```

Comment:
`.get(0)` looks bad. First, check if all the tenantIds are same. One request could have multiple items in the array but all should have same tenantId. I think this has already been implemented in one other PR.